################################################################################
SHORT DESCRIPTION: 
################################################################################
Export private Linux interfaces for xenfb2 to be built out of tree.

################################################################################
LONG DESCRIPTION: 
################################################################################

################################################################################
CHANGELOG 
################################################################################
Original Author: unknown
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>
Port to 4.14: Richard Turner <turnerr@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
When xenfb2 is not used anymore.

################################################################################
UPSTREAM PLAN
################################################################################
None this is specific to xenfb2 driver.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
xenfb2 will not build without set_phys_to_machine/zap_page_range exported in
Linux headers.

################################################################################
PATCHES 
################################################################################
Index: 4.14.5/arch/x86/xen/p2m.c
===================================================================
--- 4.14.5.orig/arch/x86/xen/p2m.c
+++ 4.14.5/arch/x86/xen/p2m.c
@@ -789,6 +789,7 @@ static int p2m_dump_open(struct inode *i
 {
 	return single_open(filp, p2m_dump_show, NULL);
 }
+EXPORT_SYMBOL_GPL(set_phys_to_machine);
 
 static const struct file_operations p2m_dump_fops = {
 	.open		= p2m_dump_open,
Index: 4.14.5/drivers/video/fbdev/Kconfig
===================================================================
--- 4.14.5.orig/drivers/video/fbdev/Kconfig
+++ 4.14.5/drivers/video/fbdev/Kconfig
@@ -2262,6 +2262,16 @@ config XEN_FBDEV_FRONTEND
 	  frame buffer driver.  It communicates with a back-end
 	  in another domain.
 
+config XEN_FBDEV2_FRONTEND
+        tristate "Xen virtual frame buffer support for XenClient"
+        depends on FB && XEN
+        select FB_CFB_COPYAREA
+        select FB_CFB_IMAGEBLIT
+        select FB_CFB_FILLRECT
+        default y
+        help
+          Hack to compile the needed functions for xenfb2
+
 config FB_METRONOME
 	tristate "E-Ink Metronome/8track controller support"
 	depends on FB
Index: 4.14.5/mm/memory.c
===================================================================
--- 4.14.5.orig/mm/memory.c
+++ 4.14.5/mm/memory.c
@@ -1620,6 +1620,7 @@ void zap_page_range(struct vm_area_struc
 	mmu_notifier_invalidate_range_end(mm, start, end);
 	tlb_finish_mmu(&tlb, start, end);
 }
+EXPORT_SYMBOL_GPL(zap_page_range);
 
 /**
  * zap_page_range_single - remove user pages in a given range
